core/src/main/java/org/bitcoinj/core/Locktime.java

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright by the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.bitcoinj.core;
+
+import org.bitcoinj.base.internal.TimeUtils;
+
+import java.time.Instant;
+import java.util.Objects;
+
+/**
+ * <p>
+ * Wrapper for transaction lock time, specified either as a block height or as a timestamp (in seconds
+ * since epoch). Both are encoded into the same long "raw value", as used in the Bitcoin protocol.
+ * The lock time is said to be "not set" if its raw value is zero.
+ * </p>
+ * <p>
+ * Instances of this class are immutable.
+ * </p>
+ */
+public class Locktime {
+    /**
+     * Wrap a raw value (as used in the Bitcoin protocol) into a lock time.
+     * @param rawValue raw value to be wrapped
+     * @return wrapped value
+     */
+    public static Locktime of(long rawValue) {
+        return new Locktime(rawValue);
+    }
+
+    /**
+     * Wrap a block height into a lock time.
+     * @param blockHeight block height to be wrapped
+     * @return wrapped block height
+     */
+    public static Locktime ofBlockHeight(int blockHeight) {
+        if (blockHeight >= THRESHOLD)
+            throw new IllegalArgumentException("block height too high: " + blockHeight);
+        return of(blockHeight);
+    }
+
+    /**
+     * Wrap a timestamp into a lock time.
+     * @param time timestamp to be wrapped
+     * @return wrapped timestamp
+     */
+    public static Locktime ofTime(Instant time) {
+        long secs = time.getEpochSecond();
+        if (secs < THRESHOLD)
+            throw new IllegalArgumentException("timestamp too low: " + secs);
+        return of(secs);
+    }
+
+    /**
+     * Construct an unset lock time.
+     * @return unset lock time
+     */
+    public static Locktime unset() {
+        return of(0);
+    }
+
+    private final long value;
+
+    // raw values below this threshold specify a block height, otherwise a timestamp in seconds since epoch
+    private static final int THRESHOLD = 500000000; // Tue Nov  5 00:53:20 1985 UTC
```

Comment:
I think this should be `public`. It's a standard Bitcoin constant and we can make it available. We could add a comment that suggests using the comparison methods instead, but I think there may be cases where this constant is legitimately needed.

Reply:
Ok, done.